feat(pie-docs): DSW-1482 update modal overview page

FayeCarter opened the pull request:
## Describe your changes (can list changeset entries if preferable)


## Author Checklist (complete before requesting a review)
- [ ] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests
- [ ] If it is a `PIE Docs` change, I have reviewed the Docs site preview
- [ ] If it is a component change, I have reviewed the Storybook preview
- [ ] If there are visual test updates, I have reviewed them properly before approving

## Reviewer checklists (complete before approving)
### Reviewer 1
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### Reviewer 2
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

xander-marjoram commented:
Should the `modifier-header-close` have a close button in it? It looks to me like a duplicate of `modifier-footer-double`.

xander-marjoram commented on apps/pie-docs/src/components/modal/overview/overview.md:
```diff
@@ -0,0 +1,424 @@
+    dont: {
+        type: usageTypes.text,
+        items: [
+            "Modal’s interrupt a user’s flow by design, so while effective when used correctly; Modal’s should be used sparingly to limit disruption to the user."
```
```suggestion
            "Modals interrupt a user’s flow by design, so while effective when used correctly; Modals should be used sparingly to limit disruption to the user."
```

xander-marjoram commented on apps/pie-docs/src/components/modal/overview/overview.md:
```diff
@@ -0,0 +1,424 @@
+        "**Close**: The close Icon button will close the modal without submitting any data.",
+        "**Main content**: Open slot for any content required.",
+        "**CTAs (optional)**: Single or dual call to action Buttons to outline the user’s next options.",
+        "**Overlay**: Screen overlay that obscures the on-page content (1.Elements/1. Backgrounds/3. Page/Overlay)."
```
What does this bit in brackets mean? It's not immediately obvious to me

FayeCarter replied:
I'm not actually sure, @nazanin 
It's for this image
<img width="800" alt="Screenshot 2023-11-24 at 11 56 12" src="https://github.com/justeattakeaway/pie/assets/24740015/7c4d7ac9-7a30-4bd2-ad95-bc0468f62936">

nazanin replied:
@xander-marjoram @FayeCarter That's the address where the token exist on design, but I actually have no idea why that piece of information exist, there was a rush to get this up and no optimization has been done on the content of modal's doc, so any recommendations and rewordings are welcome, you can delete it now, I'll track to see why it's there and an alternative

FayeCarter replied:
👍  thanks

xander-marjoram commented on apps/pie-docs/src/components/modal/overview/overview.md:
```diff
@@ -0,0 +1,424 @@
+    items: [
+        "**Back (optional)**: The chevron Icon button allows the user to return back a stage in a multi-step modal.",
+        "**Title**: Gives the users an overview of the content.",
+        "**Close**: The close Icon button will close the modal without submitting any data.",
```
```suggestion
        "**Close (optional)**: The close icon button will close the modal without submitting any data.",
```

xander-marjoram commented on apps/pie-docs/src/components/modal/overview/overview.md:
```diff
@@ -0,0 +1,424 @@
+
+#### Close
+
+Use this Header modifier when if the modal can't be dismissed.
```
I think it should be the opposite
```suggestion
Use this Header modifier when the modal can be dismissed.
```

xander-marjoram commented on apps/pie-docs/src/components/modal/overview/overview.md:
```diff
@@ -0,0 +1,424 @@
+
+#### Double
+
+A maximum of two Buttons are allowed within the Modal’s footer, with the primary Button right aligned and the lower emphasis Button on the left.
```
```suggestion
A maximum of two Buttons are allowed within the Modal’s footer, with the primary Button right-aligned and the lower-emphasis Button on the left.
```

xander-marjoram commented on apps/pie-docs/src/components/modal/overview/overview.md:
```diff
@@ -0,0 +1,424 @@
+
+#### Full screen
+
+The Wide large and optionally medium size, has the corresponding responsive option of fullscreen at narrow.
```
This sentence doesn't read very naturally to me. Is it trying to say that large modals are fullscreen at narrow viewports, and medium modals have the option of being fullscreen at narrow viewports?

FayeCarter replied:
@nazanin

FayeCarter replied:
For reference Naz, the 75% is 
'The Wide small and optionally medium size, has the corresponding responsive option of 75% width at narrow.'

nazanin replied:
We have 2 terms for describing screen sizes in component structure, wide for desktop screens and narrow for mobile and smaller ones, It's saying that a modal (large size and potentially medium) turns into a full screen modal on mobile screens, and smaller size modals will 75% with of the whole screen, I'm not the native to reword this :D

FayeCarter replied:
Thanks I'll try and updated :)

FayeCarter replied:
I have gone for 
Large modals will expand to fill the entire page width at narrow viewports.
and
Medium and small modals will expand to 75% page width at narrow viewports.
What do you think?

jamieomaguire commented on apps/pie-docs/src/components/modal/overview/sizes.json:
```diff
@@ -0,0 +1,39 @@
+  "headings": [
+    "Size",
+    "Breakpoint",
+    "Width and height"
```
Could we have separate columns for width, height, max width and max height?

FayeCarter replied:
@nazanin

nazanin replied:
Yeah that's a good point, I'll take a look at this on Monday

xander-marjoram commented on apps/pie-docs/src/components/modal/overview/overview.md:
```diff
@@ -0,0 +1,424 @@
+
+#### Top
+
+The Modal is positioned at the center of the screen by default.
```
```suggestion
The Modal is positioned at the top of the screen by default.
```

xander-marjoram commented on apps/pie-docs/src/components/modal/overview/overview.md:
```diff
@@ -0,0 +1,424 @@
+
+{% contentPageImage {
+  src: "../../../assets/img/components/modal/overview/layout-position-top.png",
+  alt: "A modal that is positioned in the middle of the screen.",
```
```suggestion
  alt: "A modal that is positioned at the top of the screen.",
```

xander-marjoram commented on apps/pie-docs/src/components/modal/overview/overview.md:
```diff
@@ -0,0 +1,424 @@
+
+#### Footer pinned = true
+
+If the actions are required to be sticky, they are pinned to the bottom of the Modal container whilst the user scrolls to indicate there is more content below until.
```
```suggestion
If the actions are required to be sticky, they are pinned to the bottom of the Modal container whilst the user scrolls to indicate there is more content below.
```

xander-marjoram commented on apps/pie-docs/src/components/modal/overview/overview.md:
```diff
@@ -0,0 +1,424 @@
+
+#### Footer pinned = false
+
+If the actions aren’t required to be sticky and placed at the bottom of the content, and the passive footer is pinned to the bottom of the Modal container whilst the user scrolls to indicate there is more content below.
```
I think this parapraph could do with rewording

FayeCarter replied:
Should it just be

'If the actions aren’t required to be sticky and are placed at the bottom of the content, the passive footer is pinned to the bottom of the Modal container whilst the user scrolls to indicate there is more content below.'?

nazanin replied:
"indicates the presence of additional content below"?

xander-marjoram commented on apps/pie-docs/src/components/modal/overview/overview.md:
```diff
@@ -0,0 +1,424 @@
+
+{% contentPageImage {
+  src: "../../../assets/img/components/modal/overview/overflow-body-content-footer-not-pinned.png",
+  alt: "A modal with no pinned footer.",
```
```suggestion
  alt: "A modal with an unpinned footer.",
```

xander-marjoram commented on apps/pie-docs/src/components/modal/overview/overview.md:
```diff
@@ -0,0 +1,424 @@
+
+{% contentPageImage {
+  src: "../../../assets/img/components/modal/overview/interaction-close-dismissible.png",
+  alt: "A modal with a dissmissible close button.",
```
```suggestion
  alt: "A modal with a dismissible close button.",
```

xander-marjoram commented on apps/pie-docs/src/components/modal/overview/overview.md:
```diff
@@ -0,0 +1,424 @@
+
+{% contentPageImage {
+  src: "../../../assets/img/components/modal/overview/interaction-close-non-dismissible.png",
+  alt: "A modal with a dissmissible close button.",
```
```suggestion
  alt: "A non-dismissible modal without a close button.",
```

xander-marjoram commented on apps/pie-docs/src/components/modal/overview/overview.md:
```diff
@@ -0,0 +1,424 @@
+
+### LTR examples
+
+Here are some examples of Buttons in left to right context:
```
```suggestion
Here are some examples of Buttons in a left-to-right context:
```

xander-marjoram commented on apps/pie-docs/src/components/modal/overview/overview.md:
```diff
@@ -0,0 +1,424 @@
+
+{% contentPageImage {
+    src: "../../../assets/img/components/modal/overview/example-ltr-stampcards.png",
+    alt: "A left to right example of a modal with form fileds in the modal body.",
```
```suggestion
    alt: "A left-to-right example of a modal with form fields in the modal body.",
```

xander-marjoram commented on apps/pie-docs/src/components/modal/overview/overview.md:
```diff
@@ -0,0 +1,424 @@
+
+### RTL examples
+
+Here are some examples of Buttons in right to left context:
```
```suggestion
Here are some examples of modals in a right-to-left context:
```

xander-marjoram commented on apps/pie-docs/src/components/modal/overview/overview.md:
```diff
@@ -0,0 +1,424 @@
+
+{% contentPageImage {
+    src: "../../../assets/img/components/modal/overview/example-rtl-stampcards.png",
+    alt: "A right to left example of a modal with form fileds in the modal body.",
```
```suggestion
    alt: "A right-to-left example of a modal with form fields in the modal body.",
```

xander-marjoram commented on apps/pie-docs/src/components/modal/overview/overview.md:
```diff
@@ -0,0 +1,424 @@
+
+{% contentPageImage {
+    src: "../../../assets/img/components/modal/overview/example-rtl-my-account.png",
+    alt: "A right to left example of a modal with form fileds in the modal body.",
```
This is the same alt text as the above image, is that correct?

xander-marjoram commented on apps/pie-docs/src/components/modal/overview/overview.md:
```diff
@@ -0,0 +1,424 @@
+
+{% contentPageImage {
+    src: "../../../assets/img/components/modal/overview/example-rtl-personal-information.png",
+    alt: "A right to left example of a modal with form fileds in the modal body.",
```
Same again

xander-marjoram commented on apps/pie-docs/src/components/modal/overview/overview.md:
```diff
@@ -0,0 +1,424 @@
+{% contentLayout %}
+  {% contentPageImage {
+      src: "../../../assets/img/components/modal/overview/narrow-modifier-footer-dual-action.png",
+      alt: "A modal with a single primary button in its footer on a mobile screen.",
```
Alt text should say two buttons?

FayeCarter replied:
thanks

FayeCarter replied:
I'd put the images backward not the alt text

xander-marjoram commented on apps/pie-docs/src/components/modal/overview/overview.md:
```diff
@@ -0,0 +1,424 @@
+
+#### Single
+
+You can use the Passive modifier for the footer in instances where the user doesn't need to take an action. This will remove the buttons at the bottom of the Modal.
```
Should this paragraph be swapped with the one for "Passive" below?

FayeCarter commented:
> A couple of questions:
> 
> 1. Should the `modifier-header-close` have a close button in it? It looks to me like a duplicate of `modifier-footer-double`.
> 2. Where are the structure images used? I can't see them on the preview for this PR or [feat(pie-docs): DSW-1482 update modal code page #1006](https://github.com/justeattakeaway/pie/pull/1006)

Hey,
1. I think you're right so I've switched them. Just to double check with @nazanin 
this was modifier-footer-double
<img width="318" alt="image" src="https://github.com/justeattakeaway/pie/assets/24740015/22f10040-0369-426e-8f8f-4ff67de01b90">
this was modifier-header-close
<img width="312" alt="image" src="https://github.com/justeattakeaway/pie/assets/24740015/c75167f5-47a1-4d11-9aa9-d1f6ebb977c2">

2. They're going to be used in the style page. I added all the images now just to make it easier but can remove them in the PR if you'd prefer?

xander-marjoram commented:
@FayeCarter Happy for the images to stay if they're going to be used in a following PR!

Do you know why `structure-wide-empty-footer.png` and `structure-wide-no-footer.png` seem to be the same picture? They sound pretty similar so this might be intentional, but it might be worth reusing the image if they're identical

jamieomaguire commented on apps/pie-docs/src/assets/styles/components/_contentImage.scss:
```diff
@@ -60,6 +59,16 @@ $contentImage-height-l--mobile:           400px;
+    background-color: f.$color-container-subtle;
+}
+
+.c-contentImage--hasPadding {
```
is hasPadding the most descriptive class name if it also includes rounded borders?

FayeCarter replied:
Any suggestions? Not really sure what else to name it. It's just this part.
`backdropHasPadding`? `gapBetweenImageAndBackdrop`?
![Screenshot 2023-11-30 at 11 24 37](https://github.com/justeattakeaway/pie/assets/24740015/e0427167-a317-4013-8880-40fdc4349882)

jamieomaguire replied:
off the top of my head `--rounded` and `--paddedBackdrop` or something (its a bit hard to tell which part of that image you mean - is the gap the pale grey space surrounding the image?

jamieomaguire replied:
alternatively, leave hasPadding but move radius to `--rounded`

jamieomaguire commented on apps/pie-docs/src/components/modal/overview/overview.md:
```diff
@@ -0,0 +1,421 @@
+    do: {
+        type: usageTypes.text,
+        items: [
+            "Use to inform users about a task can can contain information, require decisions or involve multiple tasks."
```
```suggestion
            "Use to inform users about a task can contain information, require decisions or involve multiple tasks."
```

jamieomaguire commented on apps/pie-docs/src/components/modal/overview/overview.md:
```diff
@@ -0,0 +1,421 @@
+Large modals will expand to fill the entire page width at narrow viewports.
+{% contentPageImage {
+    src: "../../../assets/img/components/modal/overview/narrow-size-full-screen.png",
+    alt: "A full screen modal on a mobile screen.",
```
```suggestion
    alt: "A fullscreen modal on a mobile screen.",
```

jamieomaguire commented on apps/pie-docs/src/components/modal/overview/overview.md:
```diff
@@ -0,0 +1,421 @@
+
+#### Center
+
+The Modal is positioned at the center of the screen by default.
```
The casing for Modal and Button seems a little inconsistent on this page. I personally prefer lowercase but please can you ensure it's the same everywhere whichever you choose?

raoufswe commented on apps/pie-docs/src/_11ty/shortcodes/componentDetailsTable.js:
```diff
@@ -59,15 +59,25 @@ const buildRow = (cells) => cells.map((cell) => {
-        <th>${heading}</th>`).join('')}
+    <table class="c-componentDetailsTable ${hasWidePadding ? 'c-componentDetailsTable-hasWidePadding' : ''}">
+    ${headings ? `<tr>${headings.map((heading, index) => {
+        const isPriorityColumn = priority && priority === index;
```
do we still need this change?
if not, we can clean it up in a follow up pr

FayeCarter replied:
I'll do it on the next one, thanks :)
